llvm/lib/ProfileData/Coverage/CoverageMapping.cpp

```diff
@@ -873,8 +874,10 @@ Error CoverageMapping::loadFunctionRecord(
       consumeError(std::move(E));
       return Error::success();
     }
-    Function.pushRegion(Region, *ExecutionCount, *AltExecutionCount,
-                        ProfileReader.hasSingleByteCoverage());
+    Function.pushRegion(
+        Region, (SingleByteCoverage && *ExecutionCount ? 1 : *ExecutionCount),
+        (SingleByteCoverage && *AltExecutionCount ? 1 : *AltExecutionCount),
+        SingleByteCoverage);
```

Comment:
The last arg will be pruned after #110966.

Reply:
I'll rewind this entirely with #110966 to make the viewer responsible to round counters.